framework/src/media/Demuxer.cpp

```diff
@@ -0,0 +1,60 @@
+/* ****************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#include <tinyara/config.h>
+#include <debug.h>
+#include "Demuxer.h"
+#ifdef CONFIG_MPEG2_TS
+#include "demux/mpeg2ts/TSDemuxer.h"
+#endif
+
+namespace media {
+
+std::shared_ptr<Demuxer> Demuxer::create(audio_container_t containerType)
+{
+	std::shared_ptr<Demuxer> demuxer;
+
+	switch (containerType) {
+#ifdef CONFIG_MPEG2_TS
```

Comment:
1 question, is demuxer is useful for another container except mpeg ts?

Reply:
yes, we can derive new demuxer class of another container format from this base class.